### PR TITLE
[Merged by Bors] - feat(cli): added proxy support during packages installation

### LINKED ISSUE #2517: [Bug]: Installation behind proxy environments
**What happened**
While installing behind a corporate proxy environment, the `fluvio` binary does not respect environment `http_proxy`, `https_proxy` variables. So, in the installation script, the command `fluvio install fluvio-cloud` fails

**Expected behavior**
Installation should succeed even behind an HTTP proxy.

**Describe the setup**
- Setting up fluvio right now

**How to reproduce it (as minimally and precisely as possible)**
1. Run the command '...'
2. Type the input '...'

**Log output**
```
# BEHIND HTTP_PROXY and HTTPS_PROXY env
curl -fsS https://packages.fluvio.io/v1/install.sh | bash
fluvio: ⏳ Downloading Fluvio 0.9.32 for x86_64-unknown-linux-musl...
fluvio: ⬇️  Downloaded Fluvio, installing...
fluvio: ✅ Successfully installed /home/intel/.fluvio/bin/fluvio-stable
fluvio: ⏳ Downloading Fluvio channel frontend 0.9.32 for x86_64-unknown-linux-musl...
fluvio: ✅ Successfully installed /home/intel/.fluvio/bin/fluvio
fluvio: ☁️ Installing Fluvio Cloud...
Current channel: stable
Error: 
   0: Http Error: Network unreachable (os error 101)
```

**Environment (please complete the following information):**
 - OS: Linux
 - Fluvio Version: stable, CLI: 0.9.32

**Additional context**
Installation behind corporate proxy environment

### PR DESCRIPTION
**Pull request by galibey**
Migrated to `isahc` crate that uses `libcurl`, which gives us more consistent package installation including support of `http_proxy` and `https_proxy` environment variables.

Fixes #2517

### CODE REVIEW

File: crates/fluvio-package-index/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "fluvio-package-index"
-version = "0.6.1"
+version = "0.6.2"
```
Comment (sehz) [inline crates/fluvio-package-index/Cargo.toml:3]: Since we are changing API, the version should be bumped to `0.7.0`

**Review by sehz — CHANGES_REQUESTED** [review]
LGTM.  A minor nit about the crate version.

**Comment by sehz** [comment]
nice use of semantic changelog!

File: crates/fluvio-package-index/Cargo.toml (hunk 1 above)
Comment (galibey) [reply crates/fluvio-package-index/Cargo.toml]: Done. `fluvio-extension-common` depends on this crate. Should I bump the patch or minor version on that case?

Comment (sehz) [reply crates/fluvio-package-index/Cargo.toml]: Looks like `fluvio-extension-common` only needs a small API from `fluvio-package-index` and shouldn't bring the entire HTTP stack.   Let's add a feature flag to `fluvio-extension-common` to make package retrieval optional.

Comment (galibey) [reply crates/fluvio-package-index/Cargo.toml]: Done

**Review by sehz — APPROVED** [review]
LGTM.  Nice job

**Pull request closed without merging**